pynbody/halo/adaptahop.py

```diff
@@ -216,8 +220,17 @@ def _get_halo_numbers_and_file_offsets(self):
                     npart = fpu.read_int()
                 self._npart[i] = npart
                 fpu.skip(1)  # skip over fortran field with ids of parts
-                self._halo_numbers[i] = fpu.read_int()
-                fpu.skip(Nskip) # skip over attributes
+
+                # AdaptaHOP might use long ints for iord only, or for every
+                # int in the code. Hide this subtlety to the user with a try catch.
+                try:
+                    ipos = fpu.tell()
+                    self._halo_numbers[i] = default_reader()
```

Comment:
Would it not be better here, and in the similar code below, to implement a `read_int_or_int64` method in `FortranFile` that inspects the field size and factors all this away where it more naturally belongs?